pkgs/ok_http/lib/src/ok_http_web_socket.dart

```diff
@@ -0,0 +1,190 @@
+                BinaryDataReceived(byteString.toByteArray().toUint8List()));
+          },
+          onClosing:
+              (bindings.WebSocket webSocket, int i, JString string) async {
+            _okHttpClientClose();
+
+            if (_events.isClosed) return;
+
+            _events.add(CloseReceived(i, string.toDartString()));
+            await _events.close();
+          },
+          onFailure: (bindings.WebSocket webSocket, JObject throwable,
+              bindings.Response response) {
+            if (_events.isClosed) return;
+
+            var throwableString = throwable.toString();
+
+            // If the throwable is:
+            // - java.net.ProtocolException: Control frames must be final.
+            // - java.io.EOFException
+            // - java.net.SocketException: Socket closed
+            // Then the connection was closed abnormally.
+            if (throwableString.contains(RegExp(
+                r'(java\.net\.ProtocolException: Control frames must be final\.|java\.io\.EOFException|java\.net\.SocketException: Socket closed)'))) {
+              _events.add(CloseReceived(1006, 'Closed abnormal'));
+              unawaited(_events.close());
+              return;
+            }
+            var error = WebSocketException(
+                'Connection ended unexpectedly $throwableString');
+            if (openCompleter.isCompleted) {
+              _events.addError(error);
+              return;
+            }
+            openCompleter.completeError(error);
+          },
+        ))));
+
+    return openCompleter.future;
+  }
+
+  @override
+  Future<void> close([int? code, String? reason]) async {
+    if (_events.isClosed) {
+      throw WebSocketConnectionClosed();
+    }
+
+    if (code != null && code != 1000 && !(code >= 3000 && code <= 4999)) {
+      throw ArgumentError('Invalid argument: $code, close code must be 1000 or '
+          'in the range 3000-4999');
+    }
+    if (reason != null && utf8.encode(reason).length > 123) {
+      throw ArgumentError.value(reason, 'reason',
+          'reason must be <= 123 bytes long when encoded as UTF-8');
+    }
+
+    unawaited(_events.close());
+
+    // When no code is provided, cause an abnormal closure to send 1005.
+    if (code == null) {
+      _webSocket.cancel();
+      return;
+    }
+
+    _webSocket.close(
+        code, reason?.toJString() ?? JString.fromReference(jNullReference));
+  }
+
+  @override
+  Stream<WebSocketEvent> get events => _events.stream;
+
+  @override
+  String get protocol => _protocol ?? '';
+
+  @override
+  void sendBytes(Uint8List b) {
+    if (_events.isClosed) {
+      throw WebSocketConnectionClosed();
+    }
+    _webSocket.send1(bindings.ByteString.of(b.toJArray()));
+  }
+
+  @override
+  void sendText(String s) {
+    if (_events.isClosed) {
+      throw WebSocketConnectionClosed();
+    }
+    _webSocket.send(s.toJString());
+  }
+
+  /// Closes the OkHttpClient using the recommended shutdown procedure.
```

Comment:
Nit (mostly because I looked this up): maybe link to https://square.github.io/okhttp/5.x/okhttp/okhttp3/-ok-http-client/index.html or where this is mentioned?

Reply:
Not proceeding as the WebSocket example has been removed.

Reply:
Done!